File: plugins/aed/scripts/test/fixtures/good_naming.ex
# AED linter fixture — Elixir naming the canon would accept.
# Expectation: zero findings of any severity.
defmodule AedFixtures.GoodNaming do
  use Ecto.Schema

  schema "customers" do
    field :first_name, :string
    field :last_name, :string
    field :email_address, :string
    field :has_a_valid_payment_method, :boolean
    field :list_of_previous_order_ids, {:array, :integer}
    field :subscription_totals_by_billing_period, :map
  end

  def lock_every_customer_account_in(list_of_customer_records_to_lock) do
    list_of_locked_customer_records =
      Enum.map(list_of_customer_records_to_lock, fn customer_record -> lock_one_customer_record(customer_record) end)

    list_of_locked_customer_records
  end

  defp lock_one_customer_record(customer_record_to_lock) do
    %{customer_record_to_lock | has_this_account_been_locked: true}
  end
end

File: plugins/aed/scripts/test/fixtures/good_naming.rb
# AED linter fixture — Ruby naming the canon would accept.
# Modelled on the canon's own Rails process manager example.
# Expectation: zero findings of any severity.
module GoodNaming
  class Customer < ApplicationRecord
    attr_reader :first_name, :last_name, :email_address

    attribute :has_a_valid_payment_method, :boolean
    attribute :is_this_an_enterprise_customer, :boolean
  end

  class UpdateCustomerPaymentAndSubscription
    attr_reader :customer_to_update, :new_payment_method_to_apply

    def initialize(customer_to_update:, new_payment_method_to_apply:)
      @customer_to_update = customer_to_update
      @new_payment_method_to_apply = new_payment_method_to_apply
    end

    def perform
      update_the_customers_payment_method && update_the_customers_subscription_status
    end

    private def update_the_customers_payment_method
      customer_to_update.update(payment_method: new_payment_method_to_apply)
    end

    private def update_the_customers_subscription_status
      list_of_subscriptions_to_reactivate = customer_to_update.subscriptions.select do |subscription_record|
        subscription_record.currently_expired?
      end
      list_of_subscriptions_to_reactivate.each { |subscription_record| subscription_record.reactivate! }
    end

    private def keep_worker_reference_alive(worker_task)
      _ = worker_task
      _ignored_result = worker_task
    end
  end
end
